src/goto-symex/symex_function.cpp

```diff
@@ -405,8 +405,18 @@ void goto_symext::symex_function_call_deref(const expr2tc &expr)
       return true;
 
     const code_type2t &ct = to_code_type(sym->type);
+
+    // check whether the number of arguments matches
     if(ct.arguments.size() != call.operands.size())
-      return true;
+    {
+      // C allows function calls as func()
+      // This happens in some multi-threaded C programs from SV-COMP
+      // i.e. void* func() instead of void* func(void *arg)
+      if(
+        !has_prefix(to_symbol2t(sym).thename.as_string(), "c:@") &&
+        !ct.arguments.size())
+        return true;
+    }
```

Comment:
Do we really need to be so conservative this far down into the verification? 

What if we simply check if there are enough arguments to call the function?

Reply:
Fixed. I agree with you.

Our C++ clang frontend already detects the illegal cases. So, we don't need to be so conservative here.